### PR TITLE
feat: bundle applicationset-controller with argocd

### LINKED ISSUE #7351: Merge ApplicationSet into Argo CD installation
# Summary

The [applicationset](https://github.com/argoproj-labs/applicationset) was created with the intention to merge it into Argo CD eventually. The project had been validated and mature enough to it should be available out of the box with Argo CD.

# Motivation

In order to improve user experience, the ApplicationSet controller should be available in Argo CD out of the box.

# Proposal

The https://github.com/argoproj-labs/applicationset repository should be moved into https://github.com/argoproj org. The Argo CD controller should include https://github.com/argoproj/applicationset as a library and use it to run controller in the the process with `argocd-application-controller`. The Argo CD installation manifests should include application set manifests

### PR DESCRIPTION
**Pull request by ishitasequeira**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Fixes #7351 

Signed off by [Ishita Sequeira](@ishitasequeira)

### CODE REVIEW

**Comment by jannfis** [comment]
I have some concerns regarding the _huge_ ApplicationSet CRD (>800kb). 

This will make self-managing Argo CD - which is a quite common pattern - quite painful. `kubectl` struggles to apply a new version of the CRD into the cluster. For example:

```shell
# Initial loading of CRD into the cluster
$ time kubectl apply -f argoproj.io_applicationsets.yaml 
customresourcedefinition.apiextensions.k8s.io/applicationsets.argoproj.io created

real    0m0.579s
user    0m0.363s
sys     0m0.069s

# Adding an annotation to the CRD in the source file
$ vi argoproj.io_applicationsets.yaml

# Apply modified file - aborting after a couple of minutes...
$ time kubectl apply -f argoproj.io_applicationsets.yaml
^C
real    11m46.482s
user    11m31.281s
sys     0m3.023s
```

So, it's obvious that things will break on Day 2.

We should either patch the CRD to have Argo CD use `replace` instead of `apply` with the appropriate sync option, or make that change directly in the source CRD.

File: docs/operator-manual/architecture.md [hunk 1]
```diff
@@ -32,3 +32,4 @@ applications and compares the current, live state against the desired target sta
 the repo). It detects `OutOfSync` application state and optionally takes corrective action. It
 is responsible for invoking any user-defined hooks for lifecycle events (PreSync, Sync, PostSync)
 
+### ApplicationSet Controller
```
Comment (chetan-rns) [inline docs/operator-manual/architecture.md:35]: Please add more information about the ApplicationSet Controller in the architecture section

**Review by jgwest — CHANGES_REQUESTED** [review]
LGTM, _but_ we probably want to wait until we have an ApplicationSet version that has https://github.com/argoproj/applicationset/issues/462 released, before we merge. So I've marked this as 'request changes' until then.

File: manifests/ha/namespace-install.yaml [hunk 2]
```diff
@@ -1,4 +1,13613 @@
...
+                                - generators
+                                type: object
+                              merge:
+                                properties:
+                                  generators:
+                                    items:
+                                      properties:
+                                        clusterDecisionResource:
+                                          properties:
+
```
Comment (alexmt) [inline manifests/ha/namespace-install.yaml]: A lot of users install Argo CD into non `argocd` namespace. Please remove hardcoded namespace from applicationset components.

**Review by alexmt — CHANGES_REQUESTED** [review]

File: manifests/base/applicationset/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - https://raw.githubusercontent.com/argoproj/applicationset/release-0.3.0/manifests/install.yaml
```
Comment (jgwest) [inline manifests/base/applicationset/kustomization.yaml:4]: I believe this should point to `https://raw.githubusercontent.com/argoproj/applicationset/master/manifests/install.yaml`

I have included my reasoning here: https://github.com/argoproj/argo-cd/issues/7351#issuecomment-1020282879

**Review by jgwest — CHANGES_REQUESTED** [review]

**Review by alexmt — APPROVED** [review]
LGTM

**Review by jgwest — APPROVED** [review]
LGTM, thanks @ishitasequeira!

**Pull request merged**